featurebyte/common/retry.py

```diff
@@ -0,0 +1,57 @@
+"""
+This module contains a decorator to retry a coroutine function
+"""
+from typing import Any, Callable, Optional, Tuple, Type, Union
+
+import asyncio
+import random
+from functools import wraps
+
+
+def async_retry(
```

Comment:
Have we considered using existing libraries such as [`backoff`](https://github.com/litl/backoff) or [`tenacity`](https://github.com/jd/tenacity) for handling retry? The interface for `backoff` seems quite similar to the implementation here.